adapters/interlocked_win32.c

```diff
@@ -1,4 +1,318 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-#include "azure_c_util/interlocked.h"
\ No newline at end of file
+#include "azure_c_util/interlocked.h"
+#include "windows.h"
+
+IMPLEMENT_MOCKABLE_FUNCTION(, int32_t, interlocked_add, volatile int32_t*, addend, int32_t, value) 
+{
+	/*Codes_SRS_INTERLOCKED_43_001 [ interlocked_add shall atomically add *addend with value and store the result in *addend.]*/
+	/*Codes_SRS_INTERLOCKED_43_032: [ interlocked_add shall return the result of the addition.]*/
+	/*Codes_SRS_INTERLOCKED_WIN32_43_001: [ interlocked_add shall call InterlockedAdd from windows.h. ]*/
+	/*Codes_SRS_INTERLOCKED_WIN32_43_002: [ interlocked_add shall return the result of the addition. ]*/
+
+	return (int32_t)InterlockedAdd((volatile LONG*)addend, (LONG)value);
```

Comment:
>(int32_t) [](start = 8, length = 9)

why is the cast necessary? Unless there's a strong reason to have the cast, remove. Here and everywhere.